### ISSUE TITLE
Add new note from template

### ISSUE THREAD

**Issue by pierreweil** [foambubble/foam#115]
It would be nice to be able to select a template when creating a new note. For example, a template called "person" could look like

```
# Note title

First name: 
Last name:
Relationship:

## Notes
```
Or similarly for a daily note. Users could be save these in a `.templates` directory for example. 

As @jevakallio mentioned, a workaround for now is to define snippets in VS Code. I noticed also that _Markdown Notes_ has a setting _New Note Template_ which allows this in a very limited form (only one template can be defined).

**Comment by SanketDG** [comment]
For anyone working on this, this can be added to Foam as a recipe.

VSCode supports [Workspace specific snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_project-snippet-scope)

We can also ship with our own set of snippets (Daily Note, TODO)

**Comment by ingalless** [comment]
Dumping some initial notes:

Templates could live in the directory `.foam/templates/`

Some things to consider (will be updated)

- [ ] Does Foam exclude `.foam` from parsing?
- [ ] Setting a daily note template to be the ID of one of the templates

**Comment by riccardoferretti** [comment]
more considerations from my conversation with @ingalless and random thoughts:
- add a new `Create new note from template..` command, which would then let select the template from the list and the file name
- shall the template also specify (optionally) in which directory to create the note? (e.g. imagine the person template saying that all people are saved in the "people" directory)
- what's the format of the template file? a `json` file similar to the snippet one? (and which could include extra data, e.g. the directory)
  - what variables should be available in the template? I like the list from @SanketDG's post above on [snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_project-snippet-scope), could be a good starting point

e.g. 
```
{
  name: "person",   // mandatory
  description: "A template for a person"
  // if no content field provided, use `<name>.md` as body
  // body follows the snippet syntax, can be string or array, for compatibility
  body: "this is the content of the template. $0",   
  path: "/path/from/repo/root" | "%CURRENT%" | undefined ,  // optional, if missing root dir is assumed, or we can use the dir from the focused note via a setting
  asCommand: true  // generates a "Create new note from template: <name>" and uses <description> for the command description
}
```

**Comment by riccardoferretti** [comment]
fixed by #346
